apex/contrib/csrc/cudnn_gbn/norm_sample.cpp

```diff
@@ -27,448 +27,453 @@
-    if (status != CUDNN_STATUS_SUCCESS) {                              \
-      _error << "CUDNN failure\nError: " << cudnnGetErrorString(status); \
-      FatalError(_error.str());                                        \
-    }                                                                  \
-}
-
-#define checkCudaErrors(status) {                                      \
-    std::stringstream _error;                                          \
-    if (status != 0) {                                                 \
-      _error << "Cuda failure\nError: " << cudaGetErrorString(status); \
-      FatalError(_error.str());                                        \
-    }                                                                  \
+// some helpers
+int64_t checkCudaError(cudaError_t code, const char* expr, const char* file, int line) {
+    if (code) {
+        printf("CUDA error at %s:%d, code=%d (%s) in '%s'", file, line, (int)code, cudaGetErrorString(code), expr);
+	return 1;
+    }
+    return 0;
 }
 
-int checkCudnnError(cudnnStatus_t code, const char* expr, const char* file, int line) {
+int64_t checkCudnnError(cudnnStatus_t code, const char* expr, const char* file, int line) {
     if (code) {
         printf("CUDNN error at %s:%d, code=%d (%s) in '%s'\n", file, line, (int)code, cudnnGetErrorString(code), expr);
         return 1;
     }
     return 0;
 }
 
-#define checkCudnnErr(...)                                                        \
-    do {                                                                          \
-        int err = checkCudnnError(__VA_ARGS__, #__VA_ARGS__, __FILE__, __LINE__); \
-        if (err) {                                                                \
-            return;                                                    \
-	}                                                                         \
-    } while (0)
-
 bool
 AllowAll(cudnnBackendDescriptor_t engine_config) {
-    (void)engine_config;
-    return false;
+  (void)engine_config;
+  return false;
 }
 
-void generateStrides(const int64_t* dimA, int64_t* strideA, int nbDims, cudnnTensorFormat_t filterFormat) {
-    // For INT8x4 and INT8x32 we still compute standard strides here to input
-    // into the cuDNN functions. We will manually scale by resizeFactor in the cpu ref.
-    if (filterFormat == CUDNN_TENSOR_NCHW) {
-        strideA[nbDims - 1] = 1;
-        for (int64_t d = nbDims - 2; d >= 0; d--) {
-            strideA[d] = strideA[d + 1] * dimA[d + 1];
-        }
-    } else {
-        // Here we assume that the format is CUDNN_TENSOR_NHWC
-        strideA[1]          = 1;
-        strideA[nbDims - 1] = strideA[1] * dimA[1];
-        for (int64_t d = nbDims - 2; d >= 2; d--) {
-            strideA[d] = strideA[d + 1] * dimA[d + 1];
-        }
-        strideA[0] = strideA[2] * dimA[2];
+void generateStrides(const int64_t* dimA, int64_t* strideA, int64_t nbDims, cudnnTensorFormat_t filterFormat) {
```

Comment:
Is it possible to simply use the Tensor's existing `.strides()` rather than relying on another helper function? AFAIK it would respect the NHWC vs. NCHW convention.

Reply:
this is how it was implemented before though, I basically based the new implementation on the old one. We could of course use strides and pass the stride tensors to the routine. Do we want to change that?